tests/main/snap-quota-memory/task.yaml

```diff
@@ -20,6 +20,15 @@ systems:
 prepare: |
   snap install go-example-webserver hello-world
 
+restore: |
+  snap remove-quota group-five || true
+  snap remove-quota group-four || true
+  snap remove-quota group-three || true
+  snap remove-quota too-small || true
+  snap remove-quota group-one || true
+  # despite removing groups, in 18.04 and 20.04, the slices stay behind
+  find /sys/fs/cgroup/ -type d -name 'snap.*.slice' -prune -ls -exec rmdir \{\} \;
```

Comment:
Should we consider this in a shared cleanup and then only do this for 18.04 and 20.04 to catch any issues from systems where they *should* have been cleaned up?

Reply:
I've only seen the problem in the 4 tests I modified in this PR

Reply:
and had to add something similar to `cleanup-state` anyway, as our service-removal-in-user-sessions really is best effort